python/mxnet/gluon/utils.py

```diff
@@ -195,6 +197,42 @@ def check_sha1(filename, sha1_hash):
     return sha1.hexdigest() == sha1_hash
 
 
+if not sys.platform.startswith('win32'):
+    # refer to https://github.com/untitaker/python-atomicwrites
+    def _replace_atomic(src, dst):
+        """Implement atomic os.replace with linux and OSX. Internal use only"""
+        try:
+            os.rename(src, dst)
+        except OSError:
+            raise OSError('os.rename failed. Might be wrong source path.')
```

Comment:
Error message should be more user friendly. Something like - "Moving downloaded temp file - %s, to %s failed. Please retry the download.".format(src, dst) .
Or something similar.